### PR TITLE
Update edit to use Progress

### PR DESCRIPTION
**Pull request by wongyx**

### CODE REVIEW

File: src/test/java/seedu/library/logic/commands/EditBookmarkDescriptorTest.java [hunk 1]
```diff
@@ -41,7 +41,7 @@ public void equals() {
         assertFalse(DESC_AMY.equals(editedAmy));
 
         // different phone -> returns false
```
Comment (zenithyap) [inline src/test/java/seedu/library/logic/commands/EditBookmarkDescriptorTest.java:43]: phone to be changed to progress

File: src/test/java/seedu/library/testutil/BookmarkBuilder.java [hunk 2]
```diff
@@ -76,8 +76,8 @@ public BookmarkBuilder withAuthor(String author) {
     /**
      * Sets the {@code Phone} of the {@code Bookmark} that we are building.
```
Comment (zhoyx) [inline src/test/java/seedu/library/testutil/BookmarkBuilder.java:77]: Replace Phone with Progress

File: src/test/java/seedu/library/model/bookmark/BookmarkTest.java [hunk 3]
```diff
@@ -73,7 +73,7 @@ public void equals() {
         assertFalse(ALICE.equals(editedAlice));
 
         // different phone -> returns false
```
Comment (zhoyx) [inline src/test/java/seedu/library/model/bookmark/BookmarkTest.java:75]: Replace phone with progress

File: src/test/java/seedu/library/logic/parser/ParserUtilTest.java [hunk 4]
```diff
@@ -80,26 +80,26 @@ public void parseTitle_validValueWithWhitespace_returnsTrimmedTitle() throws Exc
...
 
     @Test
     public void parsePhone_validValueWithWhitespace_returnsTrimmedPhone() throws Exception {
-        String phoneWithWhitespace = WHITESPACE + VALID_PHONE + WHITESPACE;
-        Phone expectedPhone = new Phone(VALID_PHONE);
-        assertEquals(expectedPhone, ParserUtil.parsePhone(phoneWithWhitespace));
+        String phoneWithWhitespace = WHITESPACE + VALID_PROGRESS + WHITESPACE;
+        Progress expectedProgress = new Progress(VALID_PROGRESS);
+        assertEquals(expectedProgress, ParserUtil.parseProgress(phoneWithWhitespace));
     }
```
Comment (zhoyx) [inline src/test/java/seedu/library/logic/parser/ParserUtilTest.java:103]: Replace phone with progress

File: src/test/java/seedu/library/logic/parser/EditCommandParserTest.java [hunk 5]
```diff
@@ -141,8 +147,8 @@ public void parse_oneFieldSpecified_success() {
         assertParseSuccess(parser, userInput, expectedCommand);
 
         // phone
```
Comment (zhoyx) [inline src/test/java/seedu/library/logic/parser/EditCommandParserTest.java:149]: Replace phone with progress

File: src/test/java/seedu/library/logic/parser/AddCommandParserTest.java [hunk 6]
```diff
@@ -87,54 +87,54 @@ public void parse_compulsoryFieldMissing_failure() {
         String expectedMessage = String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE);
 
         // missing name prefix
-        assertParseFailure(parser, VALID_TITLE_BOB + PHONE_DESC_BOB + GENRE_DESC_BOB + AUTHOR_DESC_BOB,
+        assertParseFailure(parser, VALID_TITLE_BOB + PROGRESS_DESC_BOB + GENRE_DESC_BOB + AUTHOR_DESC_BOB,
                 expectedMessage);
 
         // missing phone prefix
```
Comment (zhoyx) [inline src/test/java/seedu/library/logic/parser/AddCommandParserTest.java:93]: Replace phone with progress

File: src/test/java/seedu/library/logic/parser/AddCommandParserTest.java [hunk 7]
```diff
@@ -48,37 +48,37 @@ public void parse_allFieldsPresent_success() {
...
-        assertParseSuccess(parser, PREAMBLE_WHITESPACE + TITLE_DESC_BOB + PHONE_DESC_BOB + GENRE_DESC_BOB
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE + TITLE_DESC_BOB + PROGRESS_DESC_BOB + GENRE_DESC_BOB
                 + AUTHOR_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedBookmark));
 
         // multiple names - last name accepted
-        assertParseSuccess(parser, TITLE_DESC_AMY + TITLE_DESC_BOB + PHONE_DESC_BOB + GENRE_DESC_BOB
+        assertParseSuccess(parser, TITLE_DESC_AMY + TITLE_DESC_BOB + PROGRESS_DESC_BOB + GENRE_DESC_BOB
                 + AUTHOR_DESC_BOB + TAG_DESC_FRIEND, new AddCommand(expectedBookmark));
 
         // multiple phones - last phone accepted
```
Comment (zhoyx) [inline src/test/java/seedu/library/logic/parser/AddCommandParserTest.java:58]: Replace phone with progress

**Pull request merged**